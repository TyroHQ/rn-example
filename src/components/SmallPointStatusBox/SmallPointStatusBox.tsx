import React from "react";
import { Text } from "react-native";
import styled from "styled-components/native";
import { Placeholder } from "../Placeholder";

const SmallPointStatusBoxContainer = styled.View`
  background-color: lightgrey;
  padding: 10px;
`;

export class SmallPointStatusBox extends React.Component<{}> {
  static defaultProps: SmallPointStatusBox["props"];
  render() {
    return (
      <Placeholder>
        small point status box
        <Placeholder>Explainer text</Placeholder>
        <Placeholder>Explainer text - 2</Placeholder>
      </Placeholder>
    );
  }
}
