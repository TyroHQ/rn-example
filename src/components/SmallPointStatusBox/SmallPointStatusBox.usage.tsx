import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { SmallPointStatusBox } from "./";

export const basic = () => <SmallPointStatusBox />;
