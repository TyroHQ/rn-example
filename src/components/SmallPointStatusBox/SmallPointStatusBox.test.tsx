import { shallow } from "enzyme";
import * as usage from "./SmallPointStatusBox.usage";

describe("<SmallPointStatusBox />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
