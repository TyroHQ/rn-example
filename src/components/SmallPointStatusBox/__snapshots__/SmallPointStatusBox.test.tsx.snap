// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmallPointStatusBox /> renders well in basic mode 1`] = `
<Placeholder
  orientation="row"
>
  small point status box
  <Placeholder
    orientation="row"
  >
    Explainer text
  </Placeholder>
  <Placeholder
    orientation="row"
  >
    Explainer text - 2
  </Placeholder>
</Placeholder>
`;
